libs/cart/guides/state.md

```diff
@@ -242,5 +242,71 @@ export class AppModule {}
 
 The only argument to `forRoot` is the configuration object. For more information, see `DaffCartStateConfiguration`.
 
-<!-- TODO: add dependency injectable reducers guide once we implement it -->
+## Dependency Injectable Reducers
+
+`@daffodil/cart/state` provides mechanisms for consuming applications and libraries to modify state reduction behavior.
+
+### Purpose
+
+`@daffodil/cart/state` consumers may wish to modify the state for the `daffCart` feature in a way not explicitly provided by `@daffodil/cart/state`. A common example is adding payment info to the cart payment object in response to non-`@daffodil/cart/state` actions.
+
+#### Before vs. After
+
+Both before and after reducer slots are provided.
+
+Custom before reducers run before the native `@daffodil/cart/state` reducers. Before reducers can be used to contribute data to state with the understanding that the data may be overwritten by subsequent reducers.
+
+Custom after reducers run after the native `@daffodil/cart/state` reducers. After reducers can be used to fully control the final data in state. After reducers should take care to not malform state in a way that breaks `@daffodil/cart/state` features.
```

Comment:
I think it's better to clearly state that as long as they meet the type constraint, this shouldn't be a problem.

Reply:
I added this to the above section